node/src/components/block_synchronizer/block_acquisition.rs

```diff
@@ -478,16 +507,22 @@ impl BlockAcquisitionState {
             BlockAcquisitionState::HaveBlockHeader(header, acquired_signatures) => {
                 // we are attempting to acquire at least ~1/3 signature weight before
                 // committing to doing non-trivial work to acquire this block
-                // thus the primary thing we are doing in this state is accumulating sigs
+                // thus the primary thing we are doing in this state is accumulating sigs.
+                // We also want to ensure we've tried at least once to fetch every potential
+                // signature.
                 maybe_block_hash = Some(header.block_hash());
                 added = acquired_signatures.apply_signature(signature);
                 match validator_weights.signature_weight(acquired_signatures.have_signatures()) {
                     SignatureWeight::Insufficient => None,
                     SignatureWeight::Weak | SignatureWeight::Strict => {
-                        Some(BlockAcquisitionState::HaveWeakFinalitySignatures(
-                            header.clone(),
-                            acquired_signatures.clone(),
-                        ))
+                        if acquired_signatures.have_no_vacant() {
```

Comment:
Why do we need to enforce that no vacant signatures exist in the signature acquisition at this point? We could already have weak finality even with vacant and/or pending slots. Is this a check to ensure we *at least* asked for all signatures before moving on?

Reply:
I'd say yes, as per the updated comment above.

Reply:
Yes again :)